site/public/js/ta-grading-rubric.js

```diff
@@ -1862,15 +1822,27 @@ function onCancelComponent(me) {
 }
 
 /**
- * Called when the overall comment box get clicked (for opening / closing)
- * @param me DOM Element of the overall comment box
+ * Called when the overall comment box is changed
  */
-function onClickOverallComment(me) {
-    toggleOverallComment(true)
-        .catch(function (err) {
-            console.error(err);
-            alert('Error opening/closing overall comment! ' + err.message);
-        });
+function onChangeOverallComment() {
+    // Get the current grader so that we can get their comment from the dom.
+    var grader = getGraderId();
+    // Set the overall comment status to "Saving..."
+    $(".overall-comment-status").text("Saving...");
+    return ajaxGetOverallComment(getGradeableId(), getAnonId()).then((lastSavedOverallCommentObj) => {
+        // Get the saved version of the overall comment
+        let lastSavedOverallComment = lastSavedOverallCommentObj[grader];
+        let domComment = getOverallCommentFromDOM(grader);
+        // See if anything has changed between the dom version and the saved version of the overall comment
+        if (domComment !== lastSavedOverallComment && domComment !== undefined) {
+            // If anything has changed, save the changes.
+            return ajaxSaveOverallComment(getGradeableId(), getAnonId(), domComment);
+        }
+    }).then( () => {
```

Comment:
Let's either store the original comment as a data attribute on the comment box (or in JS somewhere) or just always save. There's not much efficiency gain making an ajax call with DB calls just to see if we should avoid making an ajax call with DB calls.

Reply:
Updated